DSCResources/MSFT_xDatabase/MSFT_xDatabase.psm1

```diff
@@ -252,26 +272,119 @@ function DeployDac([string] $databaseName, [string]$connectionString, [string]$s
         $defaultDacPacApplicationVersion = $defaultDacPacApplicationVersion
     }
 
+    Write-Verbose "DeployDac: Loading DacFx"
+
     Load-DacFx -sqlserverVersion $sqlserverVersion
 
-    $dacServicesObject = new-object Microsoft.SqlServer.Dac.DacServices ($connectionString)
+    Write-Verbose "DeployDac: Loading DacPac"
 
-    $dacpacInstance = [Microsoft.SqlServer.Dac.DacPackage]::Load($dacpacPath)
+    try
+    {        
+        $dacpacInstance = Load-Dacpac $path
+    }
+    catch
+    {
+        Write-Verbose("Unable to load dacpac, error: $_")  
+        return    
+    }    
 
     try
     {
-        $dacServicesObject.Deploy($dacpacInstance, $databaseName,$true) 
+        if($PSBoundParameters.ContainsKey('PublishProfilePath') -and  $publishProfilePath -ne "")
+        {
+            if(!(Test-Path $publishProfilePath)){
+                Write-Verbose ("Dac Publish Profile in $publishProfilePath was not found, aborting")
```

Comment:
I think so because if you specify that you want a publish profile but the file cannot be found you could be deploying something that you didn't want to (there is a separate check to see if the publish profile has been passed in or not so you can still use this without a profile)